### PR TITLE
feat: add new repro steps api

### PR DESCRIPTION
**Pull request by a7medev**
## Description of the change

Add support for the new `Instabug.setReproStepsConfig` API and deprecate the old `Instabug.setReproStepsMode`.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
## Related issues

Jira ID: MOB-13005

### Development
- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
### Code review 
- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: lib/src/modules/instabug.dart [hunk 1]
```diff
@@ -393,10 +393,32 @@ class Instabug {
...
+  @Deprecated('Use [setReproStepsConfig] instead.')
   static Future<void> setReproStepsMode(ReproStepsMode reproStepsMode) async {
     return _host.setReproStepsMode(reproStepsMode.toString());
   }
 
+  /// Sets the repro steps mode for bugs and crashes.
+  /// [bug] repro steps mode for bug reports.
+  /// [crash] repro steps mode for crash reports.
+  /// [all] repro steps mode for both bug and crash reports, when present it
+  /// overrides [bug] and [crash].
```
Comment (abdelhamid-f-nasser) [inline lib/src/modules/instabug.dart:405]: @a7medev What do you think if we add an example here, just for reference.

**Review by abdelhamid-f-nasser — CHANGES_REQUESTED** [review]
@a7medev On point PR 💪 🚀 ! 

Just one suggestion for docs. Please review it and let me know.

File: lib/src/modules/instabug.dart (hunk 1 above)
Comment (a7medev) [reply lib/src/modules/instabug.dart]: Looks good but let's remove the `all` parameter from the example since it overrides the other two.

Comment (abdelhamid-f-nasser) [reply lib/src/modules/instabug.dart]: Awesome, here's the final doc for the method.

```dart

```dart
    /// Sets the repro steps mode for bugs and crashes.
    ///
    /// [bug] repro steps mode for bug reports.
    /// [crash] repro steps mode for crash reports.
    /// [all] repro steps mode for both bug and crash reports, when present it
    /// overrides [bug] and [crash].
    ///
    /// Example:
    /// ```dart
    /// Instabug.setReproStepsConfig(
    ///  bug: ReproStepsMode.enabledWithNoScreenshots,
    ///  crash: ReproStepsMode.enabled,
    ///  );
    static Future<void> setReproStepsConfig({
  ```
  
  I've changed the value of the `crash` param to `ReproStepsMode.enabled` to make more sense.

Comment (a7medev) [reply lib/src/modules/instabug.dart]: Just one more suggestion, let's make sure the example is properly formatted and let's add the triple ` at the end of the example code.

**Review by abdelhamid-f-nasser — APPROVED** [review]

File: lib/src/modules/instabug.dart (hunk 1 above)
Comment (abdelhamid-f-nasser) [reply lib/src/modules/instabug.dart]: Last final getsuga ten.. i mean change 🤣 

```dart
    /// Sets the repro steps mode for bugs and crashes.
    ///
    /// [bug] repro steps mode for bug reports.
    /// [crash] repro steps mode for crash reports.
    /// [all] repro steps mode for both bug and crash reports, when present it
    /// overrides [bug] and [crash].
    ///
    /// Example:
    /// ```dart
    /// Instabug.setReproStepsConfig(
    ///  bug: ReproStepsMode.enabledWithNoScreenshots,
    ///  crash: ReproStepsMode.enabled,
    ///  );
    ///  ```
    static Future<void> setReproStepsConfig({
  ```

**Review by abdelhamid-f-nasser — APPROVED** [review]

**Pull request merged**